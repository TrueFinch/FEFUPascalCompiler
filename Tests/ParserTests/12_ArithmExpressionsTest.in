program a;
var
    I: float = 1;
    D: integer = 2;
    C: integer = 3;
begin
    I := (5 * D + C - (C shl 15) / ( (+8 shr 1) * (I + 2) * ((D - -10) div 10))) + C mod 10000;
    Write(I);
end.