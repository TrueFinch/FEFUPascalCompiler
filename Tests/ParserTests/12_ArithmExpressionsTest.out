                                                           a                                                             
                                                           ┼                                                             
                                                       MainBlock                                                         
                          ┌────────────────────────────────┴──────────────────────────┐                                  
                         var                                                  CompoundStatement                          
        ┌─────────────────┴─────────────────┐                                  ┌──────┴─────────────────────────┐        
   InitVarDecl       InitVarDecl       InitVarDecl                             :=                          FunctionCall  
 ┌──────┴───────┐  ┌──────┴───────┐  ┌──────┴───────┐  ┌───────────────────────┴─┐                           ┌──┴─┐      
 i  SimpleType  1  d  SimpleType  2  c  SimpleType  3  i                         +                         write  i      
      ┌─┴                ┌┴                ┌┴                               ┌────┴──────────────────┐                    
    float             integer           integer                             -                      mod                   
                                                             ┌──────────────┴───┐                ┌──┴─┐                  
                                                             +                  /                c  10000                
                                                           ┌─┴──┐    ┌──────────┴───┐                                    
                                                           *    c   shl             *                                    
                                                          ┌┴─┐     ┌─┴┐       ┌─────┴─────┐                              
                                                          5  d     c  15      *          div                             
                                                                           ┌──┴──┐      ┌─┴──┐                           
                                                                          shr    +      -    10                          
                                                                          ┌┴─┐  ┌┴─┐  ┌─┴┐                               
                                                                          +  1  i  2  d  -                               
                                                                          ┼              ┼                               
                                                                          8              10                              